cellbase-core/src/main/java/org/opencb/cellbase/core/variant/annotation/VariantAnnotationCalculator.java

```diff
@@ -669,8 +669,16 @@ private void adjustPhasedConsequenceTypes(Object[] variantArray) {
                     if (consequenceType3 != null) {
                         String referenceCodon = consequenceType1.getCodon().split("/")[0].toUpperCase();
                         // WARNING: assumes variants are sorted according to their coordinates
-                        String alternateCodon = variant0.getAlternate() + variant1.getAlternate()
-                                + variant2.getAlternate();
+                        String alternateCodon = null;
+
+                        // negative strand
+                        if ("-".equals(variant0.getStrand())) {
+                            alternateCodon = variant2.getAlternate() + variant1.getAlternate() + variant0.getAlternate();
```

Comment:
the complementary nt of the alternate should be used here, i.e.
```
VariantAnnotationUtils.COMPLEMENTARY_NT.get(variant2.getAlternate()) + VariantAnnotationUtils.COMPLEMENTARY_NT.get(variant1.getAlternate()) + VariantAnnotationUtils.COMPLEMENTARY_NT.get(variant0.getAlternate())
```

Reply:
whoops, nice catch! Fixed